doc/mgmt/Show_Running_Configuration_HLD.md

```diff
@@ -0,0 +1,495 @@
+
+## **SONiC Management Framework CLI 'Show Running Configuration'**
+
+
+
+#### High Level Design Document
+
+**Rev 0.1**
+
+#### Table of Contents
+
+-   [[List of Tables]](#list-of-tables)
+
+-   [[Revision]](#revision)
+
+-   [[About This Manual]](#about-this-manual)
+
+-   [[Scope]](#scope)
+
+-   [[Definition/Abbreviation]](#definitionabbreviation)
+
+#### List of Tables
+
+[[Table 1: Abbreviations]](#table-1-abbreviations)
+
+#### About this Manual
+
+This document states the functional requirements and high level design of the new feature to support "show running-configuration" command.
+
+#### Scope
+
+The scope of this document is limited to the functionality described in the
+requirements section.
+
+#### Definition/Abbreviation
+
+NA
+
+
+
+#### 1 Feature Overview
+
+This feature provides an infrastructure for apps to add
+their current system configuration to the output of the "show
+running-configuration" command. This command is available at the exec
+level on the SONiC management framework CLI.
+
+#### 1.1 Functional requirements
+
+The 'show running-configuration' command will display the  configuration done
+through the CLI interface and which is stored on the system.
+
+For this command to show the entire running configuration of the system,
+all apps should support configuration through CLI interface.
+
+Any configuration done using GNMI or REST or other management
+interface which is stored and has a corresponding CLI
+command will be displayed as part of this new command.
+
+The command output can be copied and pasted to be used as
+configuration on another SONiC CLI session of the same or another
+system.
+
+The order of commands displayed in the "show running-configuration" will be
+in accordance to the order of dependency between CLI views.
+
+If configuration default values are stored, they will be shown as
+part of "show running-configuration".
+
+The infrastructure will provide functionality to filter and display
+the configuration based on the CLI views.
+
+A plugin in form of callback will be provided for apps to do their own database retrieval and
+rendering. Apps will return the list of commands to the infra which will then be
+added to the final list of commands to be displayed.
+
+#### 1.2 Restrictions
+
+Apps should not use this infrastructure to add configuration to 'show running-configuration' command for which there does not exist a CLI command.
+
+System state information should not be added to this command.
+
+The ordering between commands in the same view or between views is not supported. Hence some set of configuration will not work for copy-paste operation.
```

Comment:
What does this mean? This seems serious. We need to be able to generate/paste successfully in all cases.